.github/workflows/docker-up.yml

```diff
@@ -0,0 +1,80 @@
+name: Execute `docker compose up`
+
+on:
+  workflow_call:
+    inputs:
+      parent_trigger:
+        type: string
+        required: true
+        description: The environment where to deploy. Usually `push` or `workflow_dispatch`
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+        
+      - name: Set development environment
+        env:
+          SSH_PROD_HOST: ${{ secrets.SSH_PROD_HOST }}
+          SSH_STAGE_HOST: ${{ secrets.SSH_STAGE_HOST }}
+        run: |
+          if [[ ${{ inputs.parent_trigger }} == 'workflow_dispatch' ]]; then
+            echo "DEV_ENVIRONMENT=production" >> $GITHUB_ENV
+          else
+            echo "DEV_ENVIRONMENT=staging" >> $GITHUB_ENV
+          fi
+          echo The development environment is ${{ env.DEV_ENVIRONMENT }}
+
+      - name: Get image tag
+        run: |
+          echo "RELEASE_TAG=$(git rev-parse --short ${{ github.sha }})" >> $GITHUB_ENV
+          
+      - name: Set host URI
+        env:
+          SSH_PROD_HOST: ${{ secrets.SSH_PROD_HOST }}
+          SSH_STAGE_HOST: ${{ secrets.SSH_STAGE_HOST }}
+        run: |
+          if [[ ${DEV_ENVIRONMENT} == 'production' ]]; then
+            echo "HOST_URI=${SSH_PROD_HOST}" >> $GITHUB_ENV
+          else
+            echo "HOST_URI=${SSH_STAGE_HOST}" >> $GITHUB_ENV
+          fi          
+          
+      - name: Configure SSH
+        env:
+          PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+          DEPLOY_USERNAME: ${{ secrets.DEPLOYMENT_USERNAME }}
+        run: |
+          mkdir -p ~/.ssh/
+          echo "$PRIVATE_KEY" > ~/.ssh/aws
+          chmod 600 ~/.ssh/aws
+          cat >>~/.ssh/config <<END
+          Host ${DEV_ENVIRONMENT}
+            HostName ${HOST_URI}
+            User ${DEPLOY_USERNAME}
+            IdentityFile ~/.ssh/aws
+            StrictHostKeyChecking no
+          END
+
+      - name: Log in to ECR
+        env:
+          AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          ssh ${DEV_ENVIRONMENT} "aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com"
+
+      - name: Run `docker compose up` on remote staging host
+        if: ${{ inputs.parent_trigger == 'push' }}
+        env:
+          AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          MONGODB_URI: ${{ secrets.MONGODB_URI }}
+        run: |
+          ssh ${DEV_ENVIRONMENT} MONGODB_URI="${MONGODB_URI}" HOST_URI="${HOST_URI}" RELEASE_TAG="${RELEASE_TAG}" AWS_ACCOUNT_ID="${AWS_ACCOUNT_ID}" AWS_REGION="${AWS_REGION}" docker compose -f - up -d < ./web/deploy/docker-compose.yaml
```

Comment:
@marcelovilla I can't figure out why the first environment variable [isn't being sent](https://github.com/smokestacklightnin/osm/actions/runs/11528159494/job/32094930866). Do you have any idea?

The hack solution is to put a dummy variable before it, but I'd like to know the real reason why this is happening.

Reply:
Fixed